app/controllers/posts_controller.rb

```diff
@@ -4,13 +4,13 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @posts = Post.joins(:author).where(author: { id: @user.id }).order(created_at: :desc)
-    @comments = Comment.order(created_at: :desc)
+    @comments = Comment.includes(:author).order(created_at: :desc)
   end
 
   def show
     @user = User.find(params[:user_id])
     @post = Post.find(params[:id])
-    @comments = Comment.order(created_at: :desc)
+    @comments = Comment.includes(:author).order(created_at: :desc)
   end
```

Comment:
**Changes Required**
- [ ] You will need to eagerly load the posts together with it's comments at once to handle the `n+1` problem. 
Something like `@posts = @user.posts.includes(:comments)`